code/modules/virus2/analyser.dm

```diff
@@ -86,4 +95,68 @@
 				dish = null
 				alert_noise("buzz")
 				pause = 0
+	return
+
+/obj/machinery/disease2/diseaseanalyser/Topic(href, href_list)
+	if(..())
+		return
+	if (!usr.canmove || usr.stat || usr.restrained() || !in_range(loc, usr))
+		usr.unset_machine()
+		usr << browse(null, "window=computer")
+		return
+	if(usr) usr.set_machine(src)
+
+	if(href_list["eject"])
+		for(var/obj/item/weapon/virusdish/O in src.contents)
+			if("[O.virus2.uniqueID]" == href_list["name"])
+				O.loc = src.loc
+				toscan -= O
+		src.updateUsrDialog()
+	else if(href_list["scan"])
+		for(var/obj/item/weapon/virusdish/O in src.contents)
+			if("[O.virus2.uniqueID]" == href_list["name"])
+				if(!toscan["O"])
+					toscan += O
+	else if(href_list["print"])
+		for(var/obj/item/weapon/virusdish/O in src.contents)
+			if("[O.virus2.uniqueID]" == href_list["name"])
+				PrintPaper(O)
+	else if(href_list["close"])
+		usr << browse(null, "window=computer")
+	return
+
+/obj/machinery/disease2/diseaseanalyser/attack_hand(var/mob/user as mob)
+	user.set_machine(src)
+	var/dat
```

Comment:
Since pjb didn't illustrate just replace this like with var/dat = list(), the next make sure the next line is dat+= (so all lines are added to the list), then one line before browse use 
dat = list2text(dat)